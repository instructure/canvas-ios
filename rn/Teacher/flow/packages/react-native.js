//
// This file is part of Canvas.
// Copyright (C) 2017-present  Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as
// published by the Free Software Foundation, either version 3 of the
// License, or (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.
//

// @flow

declare var console: typeof console & { disableYellowBox: boolean }

export type PanResponderGestureState = {
  stateID: number,
  moveX: number,
  moveY: number,
  x0: number,
  y0: number,
  dx: number,
  dy: number,
  vx: number,
  vy: number,
}

export type KeyboardEventData = {
  endCoordinates: {
    width: number,
    height: number,
    screenX: number,
    screenY: number,
  },
}
