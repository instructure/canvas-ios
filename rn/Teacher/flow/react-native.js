//
// This file is part of Canvas.
// Copyright (C) 2017-present  Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as
// published by the Free Software Foundation, either version 3 of the
// License, or (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.
//

// @flow
/* eslint-disable */

// define basic style prop types
// the internal ones are the best we have
import {
  type ____ViewStyleProp_Internal,
  type ____TextStyleProp_Internal,
  type ____ImageStyleProp_Internal,
} from 'react-native/Libraries/StyleSheet/StyleSheetTypes'

declare type ViewStyleProp = ____ViewStyleProp_Internal
declare type TextStyleProp = ____TextStyleProp_Internal
declare type ImageStyleProp = ____ImageStyleProp_Internal
