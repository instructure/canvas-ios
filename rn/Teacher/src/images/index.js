//
// Copyright (C) 2017-present Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, version 3 of the License.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
//

// @flow

const images = {
  course: {
    announcements: require('./course/Announcement.png'),
    assignments: require('./course/Assignment.png'),
    discussions: require('./course/Discussions.png'),
    collaborations: require('./course/Collaboration.png'),
    conferences: require('./course/Conferences.png'),
    grades: require('./course/Grades.png'),
    pages: require('./course/Pages.png'),
    people: require('./course/People.png'),
    modules: require('./course/Modules.png'),
    quizzes: require('./course/Quiz.png'),
    home: require('./course/Home.png'),
    syllabus: require('./course/Syllabus.png'),
    files: require('./course/Files.png'),
    lti: require('./course/LTI.png'),
    settings: require('./course/Settings.png'),
    placeholder: require('./course/Placeholder.png'),
    attendance: require('./course/attendance-icon.png'),
  },
  dashboard: {
    calendar: require('./dashboard/calendar.png'),
    help: require('./dashboard/help.png'),
    info: require('./dashboard/info.png'),
    warning: require('./Warning.png'),
    invite: require('./dashboard/invite.png'),
  },
  tabbar: {
    courses: require('./tabbar/courses.png'),
    inbox: require('./tabbar/inbox.png'),
    profile: require('./tabbar/profile.png'),
    staging: require('./tabbar/link.png'),
  },
  assignments: {
    calendar: require('./assignments/Calendar.png'),
  },
  speedGrader: {
    audio: require('./speedgrader/icon_audio.png'),
    video: require('./speedgrader/icon_video_clip.png'),
    page: require('./speedgrader/page.png'),
    pdf: require('./speedgrader/pdf-icon.png'),
    warning: require('./Warning.png'),
    longPress: require('./speedgrader/Longpress.png'),
    swipe: require('./speedgrader/Swipe.png'),
    submissions: {
      lti: require('./speedgrader/submission-types/LTI.png'),
      text: require('./speedgrader/submission-types/document.png'),
      document: require('./speedgrader/submission-types/document.png'),
      discussion: require('./speedgrader/submission-types/discussion.png'),
      quiz: require('./speedgrader/submission-types/quiz.png'),
      video: require('./speedgrader/submission-types/video.png'),
      audio: require('./speedgrader/submission-types/audio.png'),
      url: require('./speedgrader/submission-types/link.png'),
    },
    turnitin: {
      error: require('./Warning.png'),
      pending: require('./clock.png'),
    },
  },
  attachments: {
    complete: require('./attachments/complete-icon.png'),
    error: require('./attachments/warning-icon.png'),
  },
  mediaComments: {
    x: require('./media-comments/Close.png'),
    trash: require('./media-comments/Trash.png'),
    send: require('./media-comments/Send.png'),
  },
  files: {
    folder: require('./files/folder.png'),
    media: require('./files/media-icon.png'),
  },
  discussions: {
    rate: require('./discussions/Like.png'),
    rated: require('./discussions/Liked.png'),
  },
  canvasLogo: require('./canvas-logo.png'),
  feedback: require('./feedback.png'),
  kabob: require('./kabob.png'),
  chevronUp: require('./chevron-up.png'),
  backIcon: require('./Back-icon.png'),
  check: require('./check-white.png'),
  starFilled: require('./star-filled.png'),
  starLined: require('./star-lined.png'),
  add: require('./Add.png'),
  x: require('./x-icon.png'),
  clear: require('./Clear.png'),
  edit: require('./edit.png'),
  pencilBG: require('./Pencil-BG.jpg'),
  pickerArrow: require('./picker-arrow.png'),
  profile: require('./profile.png'),
  published: require('./publish.png'),
  unpublished: require('./unpublish.png'),
  publishedSmall: require('./publish-small.png'),
  unpublishedSmall: require('./unpublish-small.png'),
  restricted: require('./Restricted.png'),
  upArrow: require('./mini-arrow-up.png'),
  document: require('./Document.png'),
  paperclip: require('./paperclip-icon.png'),
  attachment80: require('./attachment-80.png'),
  mail: require('./Mail.png'),
  smallMail: require('./mail-small.png'),
  group: require('./group.png'),
  trash: require('./trash.png'),
  share: require('./share.png'),
  relax: require('./relax.png'),
  hamburger: require('./hamburger.png'),
  invisible: require('./invisible.png'),
}

export default (images: *)
