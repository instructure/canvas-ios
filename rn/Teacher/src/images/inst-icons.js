//
// Copyright (C) 2019-present Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, version 3 of the License.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
//

// DO NOT EDIT: this file was generated by build_icons.js
// @flow

export type InstIconName =
  'add' | 'alerts' | 'announcement' | 'arrowOpenLeft' | 'arrowOpenRight' | 'assignment' | 'audio' |
  'box' | 'calendarMonth' | 'check' | 'complete' | 'courses' | 'dashboard' | 'discussion' |
  'document' | 'email' | 'empty' | 'folder' | 'gradebook' | 'group' | 'hamburger' | 'highlighter' |
  'instructure' | 'link' | 'lock' | 'lti' | 'marker' | 'miniArrowDown' | 'miniArrowUp' | 'module' |
  'more' | 'no' | 'outcomes' | 'paint' | 'pause' | 'pdf' | 'play' | 'prerequisite' | 'question' |
  'quiz' | 'refresh' | 'rubric' | 'settings' | 'star' | 'strikethrough' | 'text' | 'trash' |
  'trouble' | 'unlock' | 'user' | 'video' | 'x'

export default function icon (name: InstIconName, type: 'line' | 'solid' = 'line') {
  return { uri: `${name}${type[0].toUpperCase()}${type.slice(1)}` }
}
