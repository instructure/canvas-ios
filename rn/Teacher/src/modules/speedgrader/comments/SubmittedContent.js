//
// This file is part of Canvas.
// Copyright (C) 2017-present  Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as
// published by the Free Software Foundation, either version 3 of the
// License, or (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.
//

// @flow

import React, { Component } from 'react'
import {
  View,
  StyleSheet,
  Image,
  TouchableOpacity,
} from 'react-native'
import colors from '../../../common/colors'
import { Title, SubTitle } from '../../../common/text'

export type SubmittedContentDataProps = {
  contentID: string,
  icon: any,
  title: string,
  subtitle: string,
}

export type SubmittedContentActionProps = {
  onPress: Function,
}

type Props
  = SubmittedContentDataProps
  & SubmittedContentActionProps
  & {
    attachmentIndex: number,
    attemptIndex: number,
    submissionID: string,
  }

export default class SubmittedContent extends Component<Props, any> {
  selectContent = () => {
    this.props.onPress(this.props.submissionID, this.props.attemptIndex, this.props.attachmentIndex)
  }

  render () {
    return (
      <TouchableOpacity
        testID={`submitted-content.item-${this.props.contentID}`}
        style={styles.row}
        onPress={this.selectContent}
      >
        <Image
          testID={`submitted-content.icon-${this.props.contentID}`}
          resizeMode='center'
          style={{
            tintColor: colors.primaryButtonColor,
            width: 18,
            height: 18,
          }}
          source={this.props.icon}
        />
        <View style={styles.textContainer} >
          <Title
            testID={`submitted-content.title-${this.props.contentID}`}
            numberOfLines={1}
            ellipsizeMode='tail'
          >
            {this.props.title}
          </Title>
          <SubTitle
            testID={`submitted-content.subtitle-${this.props.contentID}`}
            numberOfLines={1}
            ellipsizeMode='tail'
          >
            {this.props.subtitle}
          </SubTitle>
        </View>
      </TouchableOpacity>
    )
  }
}

const styles = StyleSheet.create({
  row: {
    flexDirection: 'row',
    borderWidth: 1,
    borderColor: colors.seperatorColor,
    borderRadius: 4,
    overflow: 'hidden',
    paddingVertical: 6,
    paddingHorizontal: 8,
    marginBottom: 4,
    width: 265,
  },
  textContainer: {
    flex: 1,
    marginHorizontal: 6,
  },
})
