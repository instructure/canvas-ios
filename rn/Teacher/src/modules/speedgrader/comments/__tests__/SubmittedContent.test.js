//
// This file is part of Canvas.
// Copyright (C) 2017-present  Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as
// published by the Free Software Foundation, either version 3 of the
// License, or (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.
//

/* eslint-disable flowtype/require-valid-file-annotation */

import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
import SubmittedContent from '../SubmittedContent'
import images from '../../../../images'
import explore from '../../../../../test/helpers/explore'

jest.mock('react-native/Libraries/Components/Touchable/TouchableOpacity', () => 'TouchableOpacity')

let defaultProps = {
  contentID: '1',
  icon: images.document,
  title: 'foo',
  subtitle: 'bar',
  submissionID: '1',
  attemptIndex: '0',
  attachmentIndex: '0',
  onPress: jest.fn(),
}

beforeEach(() => jest.clearAllMocks())

test('my chat bubbles render correctly', () => {
  const tree = renderer.create(
    <SubmittedContent {...defaultProps} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('tapping calls onPress with the correct parameters', () => {
  const tree = renderer.create(
    <SubmittedContent {...defaultProps} />
  ).toJSON()
  const touchable = explore(tree).selectByID('submitted-content.item-1')
  touchable && touchable.props && touchable.props.onPress()
  expect(defaultProps.onPress).toHaveBeenCalledWith('1', '0', '0')
})
