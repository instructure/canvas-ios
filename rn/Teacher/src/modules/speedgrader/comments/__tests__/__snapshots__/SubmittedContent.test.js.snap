// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`my chat bubbles render correctly 1`] = `
<TouchableOpacity
  onPress={[Function]}
  style={
    Object {
      "borderColor": "#C7CDD1",
      "borderRadius": 4,
      "borderWidth": 1,
      "flexDirection": "row",
      "marginBottom": 4,
      "overflow": "hidden",
      "paddingHorizontal": 8,
      "paddingVertical": 6,
      "width": 265,
    }
  }
  testID="submitted-content.item-1"
>
  <Image
    resizeMode="center"
    source={
      Object {
        "testUri": "../../../src/images/Document.png",
      }
    }
    style={
      Object {
        "height": 18,
        "tintColor": "#374A59",
        "width": 18,
      }
    }
    testID="submitted-content.icon-1"
  />
  <View
    style={
      Object {
        "flex": 1,
        "marginHorizontal": 6,
      }
    }
  >
    <Text
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        Array [
          Object {
            "color": "#2D3B45",
            "fontSize": 16,
            "fontWeight": "600",
          },
          undefined,
        ]
      }
      testID="submitted-content.title-1"
    >
      foo
    </Text>
    <Text
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        Array [
          Object {
            "color": "#7F91A7",
            "fontSize": 14,
          },
          undefined,
        ]
      }
      testID="submitted-content.subtitle-1"
    >
      bar
    </Text>
  </View>
</TouchableOpacity>
`;
